### ISSUE TITLE
max number of mutations that can be processed

### ISSUE THREAD

**Issue by hkmztrk** [dami82/mutSignatures#20]
Hello, I am trying to extract mutational signatures of a cohort and in total have more than 700K mutations. I cannot compute mutational signatures because it requires more than 2K GB memory.. I was wondering what was the max number of mutations have you tried the model with?

**Comment by dami82** [comment]
You have a large number of mutations, but that doesn't seem excessive. However, the numbers that really matter are number of samples and number of signatures (these define the size of your W and H matrices), not the number of mutations. If you are having issues ingesting all mutation data at once (RAM size), try to import your VCF/MAF data sample-by-sample, compute 3-nt mutation counts for each sample, aggregate, and then run the de novo extraction analysis. Also, when I worked with samples from multiple cancer types, I found more effective (in terms of results) to first split the samples by indication and then run independent indication-specific analyses. Of course, this may not fit your hypothesis.I think that the largest single-indication analysis I ran was about 2k samples (aprox 500k total mutations) vs 10 signatures (although, most indications can be resolved with 6 signatures or less).I hope this helps.Best regards,Damiano——Damiano FantiniSent from my iPhoneOn May 16, 2023, at 10:29, hkmztrk ***@***.***> wrote:﻿
Hello, I am trying to extract mutational signatures of a cohort and in total have more than 700K mutations. I cannot compute mutational signatures because it requires more than 2K GB memory.. I was wondering what was the max number of mutations have you tried the model with?
